contracts/DropFactory.sol

```diff
@@ -0,0 +1,228 @@
+        ) private returns (address super1155) {
+            
+         bytes memory super1155Bytecode = IHelper(super1155Helper).getByteCode();
+
+        bytes memory bytecodeSuper1155 = abi.encodePacked(
+            super1155Bytecode,
+            abi.encode(_collectionName, _uri, _proxyRegistry)
+        );
+
+        bytes32 salt = keccak256(
+            abi.encodePacked(block.timestamp - 2, msg.sender)
+        );
+        assembly {
+            super1155 := create2(
+                0,
+                add(bytecodeSuper1155, 0x20),
+                mload(bytecodeSuper1155),
+                salt
+            )
+            if iszero(extcodesize(super1155)) {
+                revert(0, 0)
+            }
+        }
+
+        for (uint256 i = 0; i < _itemGroupInput.length; i++) {
+            ISuper1155(super1155).configureGroup(i + 1, _itemGroupInput[i]);
+        }
+
+        ISuper1155(super1155)._transferOwnership(_owner);
+
+        return (super1155);
+    }
+
+    function createMintShop(
+        address _owner,
+        address super1155,
+        address _paymentReceiver,
+        uint256 _globalPurchaseLimit,
+        DFStorage.PoolInput[] memory _poolInput,
+        PoolConfigurationData[] memory _poolConfigurationData,
+        DFStorage.WhitelistInput[] memory _whiteListInput
+    ) private returns (address mintShop) {
+
+        bytes memory mintShopBytecode = IHelper(mintShopHelper).getByteCode();
+        bytes memory bytecodeMintShop = abi.encodePacked(
+            mintShopBytecode,
+            abi.encode(
+                _paymentReceiver,
+                _globalPurchaseLimit
+            )
+        );
+        bytes32 salt = keccak256(abi.encodePacked(block.timestamp, msg.sender));
+        assembly {
+            mintShop := create2(
+                0,
+                add(bytecodeMintShop, 0x20),
+                mload(bytecodeMintShop),
+                salt
+            )
+            if iszero(extcodesize(mintShop)) {
+                revert(0, 0)
+            }
+        }
+        ISuper1155[] memory items = new ISuper1155[](1);
+        items[0] = ISuper1155(super1155);
+
+        IMintShop(mintShop).setItems(items);
+        
+
+        for (uint256 i = 0; i < _poolInput.length; i++) {
+            _poolInput[i].collection = super1155;
+            DFStorage.PoolRequirement memory req = _poolInput[i].requirement;
+            if (req.whitelistId != 0) {
+                IMintShop(mintShop).addWhitelist(_whiteListInput[i]);
+            }
+            IMintShop(mintShop).addPool(
+                _poolInput[i],
+                _poolConfigurationData[i].groupIds,
+                _poolConfigurationData[i].issueNumberOffsets,
+                _poolConfigurationData[i].caps,
+                _poolConfigurationData[i].prices
+            );
+        }
+        bytes32 MINT = ISuper1155(super1155).getMINTPermit();
+        IMintShop(mintShop).grantMintPermit(super1155, MINT);
```

Comment:
I think this is backwards. The `Super1155` item needs to grant a permit for its `MINT` right to the `MintShop1155`.